### PR TITLE
user registration test added. Page factory implemented

### PR DESCRIPTION
**Pull request by oryabinskiy**
Can't create new user, an error appears

### CODE REVIEW

File: src/main/java/pages/AdminPage.java [hunk 1]
```diff
@@ -2,12 +2,29 @@
 
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
```
Comment (denysLystopadskyy) [inline src/main/java/pages/AdminPage.java]: Test

File: src/main/java/pages/AdminPage.java [hunk 2]
```diff
@@ -2,12 +2,29 @@
...
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
 
 /**
  * Created by oleg on 01.06.17.
  */
 public class AdminPage extends BasePage {
 
+    @FindBy(xpath = ".//*[@id='header']/div[2]/div[5]/div[1]")
```
Comment (denysLystopadskyy) [inline src/main/java/pages/AdminPage.java]: Updated Locator

File: src/main/java/pages/AdminPage.java [hunk 3]
```diff
@@ -2,12 +2,29 @@
...
+    private WebElement QuickMenu;
+    @FindBy(xpath = ".//*[@id='header']/div[2]/div[5]/div[2]/div/ul/li[3]/div/a")
```
Comment (denysLystopadskyy) [inline src/main/java/pages/AdminPage.java]: Update Locator

File: src/main/java/pages/AdminPage.java [hunk 4]
```diff
@@ -2,12 +2,29 @@
...
+    private WebElement addNewUserItem;
+    @FindBy(css = "[id*=login]")
```
Comment (denysLystopadskyy) [inline src/main/java/pages/AdminPage.java]: Is it necessary to use * in this and following cases?

File: src/main/java/pages/AdminPage.java (hunk 3 above)
Comment (denysLystopadskyy) [reply src/main/java/pages/AdminPage.java]: If you do not use text in locator, please use css

File: src/main/java/pages/AdminPage.java [hunk 5]
```diff
@@ -17,8 +34,33 @@ public String getPageUrl(){
...
+        WebElement newUserLogin = waitForElementClickable(email);
+        clearAndTypeToElement(username+getCurrentDate()+mail, newUserLogin);
+        clearAndTypeToElement(userPassword,password);
+        clearAndTypeToElement(userPassword,passwordConf);
+
+        passwordConf.submit();
+        return this;
+    }
+
+    public void clickOnElement(WebElement webElement) {
```
Comment (denysLystopadskyy) [inline src/main/java/pages/AdminPage.java:50]: It is nice to have a wrapper if you wish to update logic. In future you can do it in one place. 
As it is a general purpose method you should move it to base class or to util class

File: src/main/java/pages/AdminPage.java (hunk 3 above)
Comment (denysLystopadskyy) [reply src/main/java/pages/AdminPage.java]: Please, look through all locators and update them

File: src/main/java/pages/AdminPage.java [hunk 6]
```diff
@@ -2,12 +2,29 @@
...
+    @FindBy(xpath = ".//*[@id='header']/div[2]/div[5]/div[2]/div/ul/li[3]/div/a")
+    private WebElement addNewUserItem;
+    @FindBy(css = "[id*=login]")
+    private WebElement email;
+    @FindBy(css = "[id*=password]")
+    private WebElement password;
+    @FindBy(css = "[id*=password-conf]")
+    private WebElement passwordConf;
+    static String locator = ".//*[@id='status-messages']/ul/li";
+    @FindBy(css = "[class*=ui-dialog-titlebar-close]")
```
Comment (denysLystopadskyy) [inline src/main/java/pages/AdminPage.java]: If you use only class for locator you can use className

File: src/main/java/pages/AdminPage.java [hunk 7]
```diff
@@ -17,8 +34,33 @@ public String getPageUrl(){
...
+
+        passwordConf.submit();
+        return this;
+    }
+
+    public void clickOnElement(WebElement webElement) {
+        webElement.click();
+    }
+
+    public String getCurrentDate(){
```
Comment (denysLystopadskyy) [inline src/main/java/pages/AdminPage.java]: This will look good in some utils class

File: src/main/java/pages/BasePage.java [hunk 8]
```diff
@@ -25,4 +29,13 @@ protected void clearAndTypeToElement(String text, WebElement webElement){
         webElement.sendKeys(text);
     }
 
+    public WebElement waitForElementPresent(String locator) {
+        wait = new WebDriverWait(driver, 10);
+        return wait.until(ExpectedConditions.presenceOfElementLocated(By.xpath(locator)));
```
Comment (denysLystopadskyy) [inline src/main/java/pages/BasePage.java:34]: Minor: you can do it on line.
Does not matter in general)

File: src/main/java/pages/HomePage.java [hunk 9]
```diff
@@ -10,64 +10,47 @@
...
+    private static final String baseURL = "https://demostore.x-cart.com/";
 
-    @FindBy(css = ".input [name='login']")
-    private WebElement emailField;
+    @FindBy(css = "[class*=header_bar-sign_in]")
+    private WebElement signInButton;
 
-    @FindBy(css = ".input [name='password']")
-    private WebElement passwordField;
+    @FindBy(css="[id*=login-email]")
```
Comment (denysLystopadskyy) [inline src/main/java/pages/HomePage.java:18]: If it is id, you should use id, not css locator. Please check all the places.

**Review by denysLystopadskyy — CHANGES_REQUESTED** [review]

**Comment by oryabinskiy** [comment]
Locators updated. Screenshots added

**Pull request closed without merging**